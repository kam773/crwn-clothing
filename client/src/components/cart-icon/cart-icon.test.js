import React from 'react';
import { shallow } from 'enzyme';
import CartIcon from './cart-icon.component';
import { ShoppingIcon, ItemCountContainer } from './cart-icon.styles';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

const initialState = {
  cart: {
    cartItems: [1, 2, 3],
    hidden: true,
  },
  itemCount: 0,
};
const mockStore = configureStore();
let store;
let container;

describe('create snapshot', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<CartIcon store={store} />);
  });
  it('capture snapshot of CartItem', () => {
    const wrapper = renderer.create(container).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('render connected component', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<CartIcon store={store} />);
  });
  it('render cart icon', () => {
    expect(container.find(ShoppingIcon)).toBeDefined();
  });
  it('render count container', () => {
    expect(container.find(ItemCountContainer)).toBeDefined();
  });
});
