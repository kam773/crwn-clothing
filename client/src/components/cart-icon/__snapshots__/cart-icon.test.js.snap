// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot capture snapshot of CartIcon 1`] = `null`;

exports[`create snapshot capture snapshot of CartItem 1`] = `
<div
  className="sc-bdVaJa kUrIC"
  onClick={[Function]}
>
  <svg
    className="sc-bwzfXH cKSSZR"
  >
    shopping-bag.svg
  </svg>
  <span
    className="sc-htpNat fFVhJp"
  >
    NaN
  </span>
</div>
`;
